pkg/partition/consistent/hashing.go

```diff
@@ -33,58 +41,145 @@ func (th rkvHash) Hash(key []byte) uint64 {
 	return xxhash.Sum64(key)
 }
 
-type KV struct {
-	key, value string
+type HashingManager interface {
+	GetSyncNodes(key []byte) ([]Node, error)
+	GetAsyncNodes(key []byte) ([]Node, error)
+	GetNodes(key []byte) ([]string, error)
+}
+
+type SyncHashingManager struct {
+	hasing ConsistentHashing
+	count  int
+}
+
+func NewSyncHashingManager(hashingType constants.ConsistentHashingType, nodes []RkvNode, count int) SyncHashingManager {
+	h := Factory(hashingType)
+	for _, node := range nodes {
+		h.AddNode(node)
+	}
+	return SyncHashingManager{hasing: h, count: count}
+}
+
+func (shm SyncHashingManager) GetSyncNodes(key []byte) ([]Node, error) {
+	return shm.hasing.LocateNodes(key, shm.count), nil
 }
 
-type HashingWithAzandRemote struct {
+func (shm SyncHashingManager) GetAsyncNodes(key []byte) ([]Node, error) {
+	return nil, nil
+}
+
+func (shm SyncHashingManager) GetNodes(key []byte) ([]string, error) {
```

Comment:
is GetNodes getting back ONLY sync nodes?

Reply:
Yes.